Doc/c-api/refcounting.rst

```diff
@@ -119,3 +119,32 @@ The following functions or macros are only for use within the interpreter core:
 :c:func:`_Py_Dealloc`, :c:func:`_Py_ForgetReference`, :c:func:`_Py_NewReference`,
 as well as the global variable :c:data:`_Py_RefTotal`.
 
+.. _immortal-objects:
+
+Immortal Objects
+================
+
+"Immortal" objects are those that are expected to never be deallocated
```

Comment:
never be deallocated?  even on main interpreter Py_Finalize()?  It seems like we should kill the immortals at that point.

Reply:
As I noted to Victor, it shouldn't be a problem for us to address immortal objects directly in `Py_Finalize()`.  The question of dealing with static objects during runtime finalization is a separately question; one worth addressing.